src/include/kernel/ext_manifest.h

```diff
@@ -0,0 +1,76 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ * Author: Karol Trzcinski <karolx.trzcinski@linux.intel.com>
+ */
+
+/*
+ * Extended manifest is a place to store metadata about firmware, known during
+ * compilation time - for example firmware version or used compiler.
+ * Given information are read on host side before firmware startup.
+ * This part of output binary is not signed.
+ *
+ * To add new content to ext_man, in firmware code define struct which starts
+ * with ext_man_elem_head followed by usage dependent content and place whole
+ * struct in "fw_metadata" section. Moreover kernel code should be modified to
+ * properly read new packet.
+ *
+ * Extended manifest is designed to be extensible. In header there is a field
+ * which describe header length, so after appending some data to header then it
+ * can be easily skipped by device with older version of this header.
+ * Unknown ext_man elements should be just skipped by host,
+ * to be backward compatible. Field `ext_man_elem_header.elem_size` should be
+ * used in such a situation.
+ */
+
+#ifndef __KERNEL_EXT_MANIFEST_H__
+#define __KERNEL_EXT_MANIFEST_H__
+
+#include <sof/common.h>
+#include <stdint.h>
+
+/* In ASCII `XMan` */
+#define EXT_MAN_MAGIC_NUMBER	0x6e614d58
+
+/* Build u32 number in format MMmmmppp */
+#define EXT_MAN_BUILD_VERSION(MAJOR, MINOR, PATH) ((uint32_t)( \
+	((MAJOR) << 24) | \
+	((MINOR) << 12) | \
+	(PATH)))
+
+/* check extended manifest version consistency */
+#define EXT_MAN_VERSION_INCOMPATIBLE(host_ver, cli_ver) ( \
+	((host_ver) & GENMASK(31, 24)) != \
+	((cli_ver) & GENMASK(31, 24)))
+
+/* used extended manifest header version */
+#define EXT_MAN_VERSION		EXT_MAN_BUILD_VERSION(1, 0, 0)
+
+/* struct size alignment for ext_man elements */
+#define EXT_MAN_ALIGN 16
+
+/* extended manifest header, deleting any field breaks backward compatibility */
+struct ext_man_header {
+	uint32_t magic;		/**< identification number, */
+				/**< EXT_MAN_MAGIC_NUMBER */
+	uint32_t full_size;	/**< [bytes] full size of ext_man, */
+				/**< (header + content + padding) */
+	uint32_t header_size;	/**< [bytes] makes header extensionable, */
+				/**< after append new field to ext_man header */
+				/**< then backward compatible won't be lost */
+	uint32_t header_version; /**< value of EXT_MAN_VERSION */
+				/**< not related with following content */
+	uint8_t elements[];	/**< list of ext_man_elem_* elements */
+} __packed;
+
+/* Now define extended manifest elements */
+
+/* extended manifest element header */
+struct ext_man_elem_header {
+	uint32_t type;		/**< EXT_MAN_ELEM_* */
+	uint32_t elem_size;	/**< in bytes, including header size */
+	uint8_t blob[];		/**< type dependent content */
```

Comment:
Whats the max size here of blob ? It can be a union if needed to reserve the space (given recent XCC/GCC differences in spacing). Btw, not super keen on this be undefined since IIIUC this will be parsed by the kernel ? @plbossart any comment on this ?

Reply:
size of `blob` is here 0, `blob` keyword is actually unused in source code, because it is only indication for developer to remember that after `ext_man_elem_header` are located `type` dependent content.

Reply:
Other structures use "align" for the same purpose?

Reply:
If it's unused then we dont need it. You can use a comment instead.

Reply:
So I put changes to kernel in https://github.com/thesofproject/linux/pull/1903

Reply:
This change is quite irrelevant, so make small fix for this, but I think that it shouldn't be more changes in header structures because corresponding kernel from kernel side has been already merged.